plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMigrateProcedure.java

```diff
@@ -223,18 +234,113 @@ public void testMigrateUnsupportedColumnType()
         String hiveTableName = "hive.tpch." + tableName;
         String icebergTableName = "iceberg.tpch." + tableName;
 
-        assertUpdate("CREATE TABLE " + hiveTableName + " AS SELECT timestamp '2021-01-01 00:00:00.000' x", 1);
+        assertUpdate("CREATE TABLE " + hiveTableName + " (x tinyint)");
+        assertUpdate("INSERT INTO " + hiveTableName + " (x) values 11,12,13", 3);
+        assertQuery("SELECT count(*) FROM " + hiveTableName, "SELECT 3");
 
-        assertQueryFails(
-                "CALL iceberg.system.migrate('tpch', '" + tableName + "')",
-                "\\QTimestamp precision (3) not supported for Iceberg. Use \"timestamp(6)\" instead.");
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '%s')".formatted(tableName), "Type not supported for Iceberg: tinyint");
+
+        assertQuery("SELECT count(*) FROM " + hiveTableName, "SELECT 3");
+        assertQueryFails("SELECT count(*) FROM " + icebergTableName, "Not an Iceberg table: .*");
+
+        assertUpdate("DROP TABLE " + hiveTableName);
+    }
+
+    @Test(dataProvider = "timestampPrecisions")
+    public void testMigrateColumnTypeTimestampOnFormatORC(HiveTimestampPrecision precision)
```

Comment:
Could you consider merging these 3 test methods into one with data provider?

Reply:
I tried but I think the code is more readable in the state as now. I would left it as it's now.